protocols/zipkinv2/encoding.go

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zipkinv2
+
+import (
+	"encoding/json"
+
+	zipkinmodel "github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/proto/zipkin_proto3"
+
+	"go.opentelemetry.io/collector/protocols/encoding"
+	"go.opentelemetry.io/collector/protocols/models"
+)
+
+var (
+	_ encoding.TracesEncoder = (*Encoder)(nil)
+	_ encoding.TracesDecoder = (*Encoder)(nil)
+)
+
+type Encoder struct {
```

Comment:
map[encoding.Type]Encoder

Reply:
I think it's possible but will require some helpers. Let's punt on it for now though as it doesn't impact the external API so can be easily tidied up later.